src/registry/Registry.ts

```diff
@@ -14,13 +16,140 @@ export class Registry {
 	readonly relayChain: RelayChains;
 	readonly currentRelayRegistry: ChainInfo;
 	readonly xcAssets: XCMChainInfoRegistry;
+	cache: ChainInfoRegistry;
 
 	constructor(specName: string, opts: AssetsTransferApiOpts) {
 		this.specName = specName;
 		this.registry = parseRegistry(opts);
 		this.relayChain = findRelayChain(this.specName, this.registry);
 		this.currentRelayRegistry = this.registry[this.relayChain];
 		this.xcAssets = this.registry.xcAssets;
+		this.cache = {
+			polkadot: {},
+			kusama: {},
+			westend: {},
+			xcAssets: {
+				polkadot: [],
+				kusama: [],
+			},
+		};
+	}
+
+	private initializeCurrentChainIdCache() {
```

Comment:
This can be called directly in the constructor so you dont have to call it anywhere else, and it will only be called once.

Reply:
very good point. Ill make it so we call this in the constructor instead